dali/pipeline/operators/operator.h

```diff
@@ -213,6 +233,100 @@ class Operator : public OperatorBase {
   template <typename B = Backend>
   typename std::enable_if<!std::is_same<B, GPUBackend>::value>::type
   SyncHelper(int /*unused*/, Workspace<B> */*unused*/) {}
+
+  template <typename B = Backend>
+  typename std::enable_if<std::is_same<B, GPUBackend>::value>::type
+  Flatten(Workspace<Backend> *ws) {
+    seq_sizes_.clear();
+    seq_sizes_.resize(input_sets_, 0);
+    for (int i = 0; i < input_sets_; ++i) {
+      CUDA_CALL(cudaStreamSynchronize(ws->stream()));
+      auto &input = ws->template MutableInput<Backend>(i);
+
+      const std::vector<Dims>& old_shapes = input.shape();
+      DALITensorLayout layout = input.GetLayout();
+      if (layout == DALI_NFHWC || layout == DALI_NFCHW) {
+        // size of seq is the first dim in each tensor
+        seq_sizes_[i] = old_shapes[0][0];
+
+        std::vector<Dims> new_shapes;
+        for (const Dims &old_shape : old_shapes) {
+          // getting only the need 3 dimensions
+          Dims new_shape(old_shape.begin() + 1, old_shape.end());
+          for (int s = 0; s < seq_sizes_[i]; ++s) {
+            new_shapes.emplace_back(new_shape);
+          }
+        }
+        input.Resize(new_shapes);
+        if (layout == DALI_NFHWC) {
```

Comment:
and probably something like `GetElementLayout(...)`

Reply:
Done